syntax = "proto3";
package zmq_if;
import "keyword_list.proto";
enum Encode {
	UTF8 = 0;
	UCS2 = 1;
	GBK = 2;
}

message HitPosition{
	int64 offset = 1;
	bytes context = 2;
  	Encode encode_hint = 3;
  	KeywordItem keyword = 4;
}
import "task.proto";
message CheckResult {
	Task task = 1;
	repeated HitPosition hit_positions = 2;
  	int32 keyword_serial = 3;
}